src/generator.js

```diff
@@ -0,0 +1,37 @@
+const P = require('bluebird')
+const path = require('path')
+const mkdirp = P.promisify(require('mkdirp'))
+
+const ls = require('./lib/list_files')
+const fork = require('./lib/fork_job')
+const os = require('os')
+
+module.exports = async (data, templatePath, outputPath) => {
+  // normalize the path
+  templatePath = path.resolve(templatePath)
+  outputPath = path.resolve(outputPath)
+
+  // create output folder if needed
+  await mkdirp(outputPath)
+
+  // generate a list of all files in this template folder
+  const files = await ls(templatePath)
+
+  // create one worker for each file
+  await P.map(files, async (file) => {
```

Comment:
This is a great starting point, but keep in mind we will need a way to compose/nest templates/block helpers. i.e, for the REST API, we can write a single block helper implementing  `_makeAuthRequest()`, and then use it in a separate `rest2.js.hbs` template file. This should scale to nesting arbitrary templates inside of each other (no need to go deeper than 2 levels imo, so root + 1 more), not just block helpers. This may require a pre-rendering stage.

Reply:
yeah i kept that in mind and i think i got a very elegant idea on how to solve that, will let you know when i get it down!